core/storage/blog/gae_models.py

```diff
@@ -0,0 +1,335 @@
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing the blog data models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import python_utils
+import utils
+(base_models, user_models) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.user])
+
+datastore_services = models.Registry.import_datastore_services()
+
+# Constants used for generating new ids.
+_MAX_RETRIES = 10
+_RAND_RANGE = 127 * 127
+
+
+class BlogPostModel(base_models.BaseModel):
+    """Model to store blog content
+
+    The id of instances of this class has the form
+        [author_id].[generated_string]
+    """
+
+    # The ID of the user the blog is authored by.
+    author_id = datastore_services.StringProperty(required=True)
+    # Title of the blog.
+    title = datastore_services.StringProperty(indexed=True, required=True)
+    # Content of the blog.
+    content = datastore_services.TextProperty(indexed=False, required=True)
+    # The url fragment of the blog.
+    url_fragment = (
+        datastore_services.StringProperty(indexed=True, required=True))
```

Comment:
Could you please explain a bit more about what this is? Specifically the comment should answer the question: "why is it OK to query by this value?"

Reply:
If the user directly enters the URL of the blog-editor or that of the blog-post page, we should be able to get the data by the URL fragment for the blog post.